### PR TITLE
Remove batch_size of bulk API from tests & refactor BWC version check

### PR DESCRIPTION
**Pull request by chishui**
### Description
1. Remove the use of `batch_size` of bulk API from tests
2. Update BWC tests to use `batch_size` in processor
3. Refactor BWC version check in gradle file
 
### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/14283

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/neural-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "processors": [
     {
       "sparse_encoding": {
-        "model_id": "%s",
+        "model_id": "%s",{{batch_size}}
```
Comment (zhichao-aws) [inline qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json:6]: Can we use the same style of filling the model_id?

File: src/test/resources/processor/PipelineConfiguration.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "processors": [
     {
       "text_embedding": {
-        "model_id": "%s",
+        "model_id": "%s",{{batch_size}}
```
Comment (zhichao-aws) [inline src/test/resources/processor/PipelineConfiguration.json:6]: same as above

File: src/test/resources/processor/SparseEncodingPipelineConfiguration.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "processors" : [
     {
       "sparse_encoding": {
-        "model_id": "%s",
+        "model_id": "%s",{{batch_size}}
```
Comment (zhichao-aws) [inline src/test/resources/processor/SparseEncodingPipelineConfiguration.json:6]: same as above

**Review by zhichao-aws — COMMENTED** [review]
Thanks for the refactor on the versionsBelow check logic.

File: qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json (hunk 1 above)
Comment (chishui) [reply qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json:6]: this is an optional parameter, sometimes, we just don't use it, so we can replace it with empty string

Comment (zhichao-aws) [reply qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json:6]: For this case can we set it as 1 (the default value)?

File: qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRestartUpgradeRestTestCase.java [hunk 4]
```diff
@@ -93,13 +93,25 @@ protected void createPipelineForTextImageProcessor(final String modelId, final S
         createPipelineProcessor(requestBody, pipelineName, modelId);
     }
 
-    protected void createPipelineForSparseEncodingProcessor(final String modelId, final String pipelineName) throws Exception {
+    protected void createPipelineForSparseEncodingProcessor(String modelId, String pipelineName, Integer batchSize) throws Exception {
```
Comment (zane-neo) [inline qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRestartUpgradeRestTestCase.java:96]: Looks like modelId and pipelineName can still be final.

File: qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRollingUpgradeTestCase.java [hunk 5]
```diff
@@ -124,13 +124,25 @@ protected String uploadSparseEncodingModel() throws Exception {
         return registerModelGroupAndGetModelId(requestBody);
     }
 
-    protected void createPipelineForSparseEncodingProcessor(String modelId, String pipelineName) throws Exception {
+    protected void createPipelineForSparseEncodingProcessor(String modelId, String pipelineName, Integer batchSize) throws Exception {
```
Comment (zane-neo) [inline qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRollingUpgradeTestCase.java:127]: Looks like repeating code, is there any opportunity to optimize this?

File: qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json (hunk 1 above)
Comment (chishui) [reply qa/rolling-upgrade/src/test/resources/processor/PipelineForSparseEncodingProcessorConfiguration.json:6]: That also works

**Review by zane-neo — APPROVED** [review]

**Pull request merged**